library/summary.ml

```diff
@@ -41,17 +42,20 @@ let name_of_summary name =
   try List.assoc name !summary_names
   with Not_found -> "summary name not found"
 
-let declare_summary sumname decl =
+let declare_typed_summary sumname decl =
   let hash = String.hash sumname in
-  let () = if Int.Map.mem hash !summaries then
+  if Int.Map.mem hash !summaries then begin
```

Comment:
Don't you like explicit state effect? This change looks completely spurious to me...